akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -126,6 +127,29 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         errorCode should ===(ErrorCode.COMPRESSION_ERROR)
       }
 
+      "GOAWAY when the response has headers mid stream" in new TestSetup with NetProbes with Http2FrameHpackSupport {
+        val streamId = 0x1
+        emitRequest(streamId, Get("/"))
+        expectDecodedHEADERS(streamId, endStream = true)
+
+        sendHEADERS(streamId, endStream = false, Seq(
+          RawHeader(":status", "200"),
+          RawHeader("content-type", "application/octet-stream")
+        ))
+
+        val response = expectResponse()
+        response.entity shouldBe a[Chunked]
+
+        sendDATA(streamId, endStream = false, ByteString("asdf"))
+        sendHEADERS(streamId, endStream = false, Seq(RawHeader("X-mid-stream", "badvalue")))
+
+        // FIXME need something like the Http2ServerSpec.expectWindowUpdate() here
+        expectFrameFlagsStreamIdAndPayload(FrameType.WINDOW_UPDATE)
```

Comment:
Is there a better way to do this (less fragile)?

Reply:
It's triggered by `sendDATA` so you could at least pull it up a line. The infrastructure in Http2ServerSpec (`pollForWindowUpdates`) would probably also work here but we haven't pulled it out so far.

Reply:
I'll leave it like this for now (but pulled it up next to the sendDATA).